### PR TITLE
 fix: compare properties of Invalid types

### PR DESCRIPTION
**Pull request by alex-savchuk**
fix: compare properties of Invalid types #1643

### CODE REVIEW

**Comment by alex-savchuk** [comment]
Problem was in processing of "github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name"
Both of these properties were not available -> comparison of values of "Invalid" type -> error is reported.

Fixed:
* adding check for such situation and returning false when both compared values are Invalid (please confirm this assumption is correct)
* Error texts are updated.
* Test is updated.

**Review by KnisterPeter — APPROVED** [review]
:+1:

File: pkg/exprparser/interpreter.go [hunk 1]
```diff
@@ -366,8 +366,16 @@ func (impl *interperterImpl) compareValues(leftValue reflect.Value, rightValue r
 
 		return impl.compareNumber(leftValue.Float(), rightValue.Float(), kind)
 
+	case reflect.Invalid:
+		if rightValue.Kind() == reflect.Invalid {
+			return false, nil
```
[the remark this answers is not in the document]
Comment (alex-savchuk) [reply pkg/exprparser/interpreter.go:371]: I assumed that Invalid is not equal to any other value even another Invalid. So I used false.

Not a problem, I could change it to true.

File: pkg/exprparser/interpreter.go [hunk 2]
```diff
@@ -366,8 +366,16 @@ func (impl *interperterImpl) compareValues(leftValue reflect.Value, rightValue r
...
+		}
+
+		// not possible situation - params are converted to the same type in code above
+		return nil, fmt.Errorf("Compare params of Invalid type: left: %+v, right: %+v", leftValue.Kind(), rightValue.Kind())
```
[the remark this answers is not in the document]
Comment (alex-savchuk) [reply pkg/exprparser/interpreter.go:375]: nil here means any. I think, as we return here an error it's not important. 

So I could change it to false.

[the remark this answers is not in the document]
Comment (alex-savchuk) [reply pkg/exprparser/interpreter.go:375]: As I understand, such case (comparison of node values of different types) is covered by code above. There we translate  both to numbers if they are not yet.

At this point we have either both are Invalid (and it's covered by IF above) or something impossible happens.
That's why I commented return here as "not possible situation".

File: pkg/exprparser/interpreter.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex-savchuk) [reply pkg/exprparser/interpreter.go]: Done

**Review by KnisterPeter — APPROVED** [review]

**Pull request merged**